Adjusts global var generation.

PsiOmegaDelta opened the pull request:
The global var generation script now outputs each entry in the global var list on individual lines, in a sorted manner, as well as with consistent newlines for all OSes. Also prints current md5.
Most of this is in hope to make casual merging easier.

Adds a from_file() macro to go with the to_file macro(). Expected to be used with preference settings, etc.

PsiOmegaDelta committed:
Adjusts global var generation.

The global var generation script now outputs each entry in the global var list on individual lines, as well as consistent newlines for all OSes. Also prints current md5.
Adds a from_file() macro to go with the to_file macro(). Expected to be used with preference settings, etc.

mwerezak commented:
So will people now need to copypaste hexstrings into travis.yml whenever they add/remove a global variable?

Techhead0 commented:
Additionally, I never liked the bracket Travis test. There are several uses besides outputting to chat, and someone has to deal with it every time they mess with savefiles or bitshifts.

We don't have a test for `loc =` or `dir =`, or for directly setting other vars with handlers. And those have less uses than `<<`.

PsiOmegaDelta commented:
> So will people now need to copypaste hexstrings into travis.yml whenever they add/remove a global variable?

That would be a consequence of this, aye. But if one add/remove global vars often enough for it to be overly annoying then perhaps one use them a bit too much.

> We don't have a test for loc = or dir =, or for directly setting other vars with handlers. And those have less uses than <<

loc is doable to also check for but dir, density, opacity, and friends is a little trickier because they may be assigned as part of a type definition.

> Additionally, I never liked the bracket Travis test. There are several uses besides outputting to chat, and someone has to deal with it every time they mess with savefiles or bitshifts.

Bitshifts is a valid point, but for everything else there is https://github.com/Baystation12/Baystation12/blob/dev/code/_macros.dm#L79.

PsiOmegaDelta commented:
Optionally I can just continue keeping these things up-to-date myself, and pad my commit count :P
And if this turns out to bite me in the butt it's not too tricky to revert parts or all of this.

PsiOmegaDelta commented:
Removed global var hash check.
Added a from_file() macro, to go with the to_file() macro. These are now the expected methods to use when dealing with file data, such as character preferences. Not strictly needed but added to aid with consistency.

PsiOmegaDelta commented:
This will soon have been approved for 3 weeks.
